fireant/slicer/dimensions.py

```diff
@@ -0,0 +1,136 @@
+from fireant.utils import immutable
+
+from .base import SlicerElement
+from .exceptions import QueryException
+from .filters import (
+    BooleanFilter,
+    ContainsFilter,
+    ExcludesFilter,
+    RangeFilter,
+    WildcardFilter,
+)
+from .intervals import (
+    NumericInterval,
+    daily,
+)
+
+
+class Dimension(SlicerElement):
+    """
+    The `Dimension` class represents a dimension in the `Slicer` object.
+    """
+
+    def __init__(self, key, label=None, definition=None):
+        super(Dimension, self).__init__(key, label, definition)
+
+
+class BooleanDimension(Dimension):
+    """
+    This is a dimension that represents a boolean true/false value.  The expression should always result in a boolean
+    value.
+    """
+
+    def __init__(self, key, label=None, definition=None):
+        super(BooleanDimension, self).__init__(key=key,
+                                               label=label,
+                                               definition=definition)
+
+    def is_(self, value):
+        return BooleanFilter(self.definition, value)
```

Comment:
Please add a docstring. I'm not sure what 'is_' this does.

Reply:
will add a doc string